tests/golden_tests/compile_file/scrutinee_reconstruction.hvm

```diff
@@ -0,0 +1,8 @@
+data Option = (Some val) | None
+
+Option.or = @a @b match a {
```

Comment:
Could you add another test for reconstructing the matched variable, but in a scenario where both the variable itself and its argument are used? Something like
```
@a match a {
  Ctr: (foo a a.field)
}
```
The expected result should be that only the field is `dup`ed and `a` is reconstructed inside the match case body